src/mqtt/publishing/publisher.cr

```diff
@@ -22,6 +32,15 @@ module PlaceOS::MQTT
       State.new(key, payload)
     end
 
+    def self.payload(value, timestamp : Time = self.timestamp)
+      value = "null" if value.nil?
+      %({"time":#{timestamp.to_unix_ms},"value":#{value}})
```

Comment:
This is a little fragile. It works for the current setup, but the JSON encoding is split across a few areas. Not essential, but could setup a struct with appropriate JSON annotations to emit null’s and provide a direct passthrough for the values if they’re pre-encoded.

Reply:
I think it just doesn't send a value if there is not one (no point sending `null`)
but otherwise it's sending JSON strings

Reply:
Fixed in #3